streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -77,7 +77,8 @@ object Checkpoint extends Logging {
   }
 
   /** Get checkpoint files present in the give directory, ordered by oldest-first */
-  def getCheckpointFiles(checkpointDir: String, fs: FileSystem): Seq[Path] = {
+  def getCheckpointFiles(checkpointDir: String, fsOption: Option[FileSystem] = None): Seq[Path] = {
```

Comment:
BTW, this file has to change anyways in the attempt to make the semantics of `read` more clear.